std/string.d

```diff
@@ -4722,46 +4722,76 @@ string[string] abbrev(string[] values) @safe pure
  * line is returned.
  *
  * Params:
- *    str = string to be analyzed
+ *    str = string or InputRange to be analyzed
  *    tabsize = number of columns a tab character represents
  *
  * Returns:
  *    column number
  */
 
-size_t column(S)(S str, in size_t tabsize = 8) @safe pure @nogc if (isSomeString!S)
+size_t column(Range)(Range str, in size_t tabsize = 8)
+    if (!isStaticArray!Range &&
+        (is(Range : const  char[]) ||
+         is(Range : const wchar[]) ||
+         is(Range : const dchar[]) ||
+         isInputRange!Range &&
+             (is(Unqual!(ElementEncodingType!Range) ==  char) ||
+              is(Unqual!(ElementEncodingType!Range) == wchar) ||
+              is(Unqual!(ElementEncodingType!Range) == dchar))
+        )
+       )
 {
-    import std.uni : lineSep, paraSep;
-
-    size_t column;
+    static if (is(Unqual!(ElementEncodingType!Range) == char))
+    {
+        // decoding needed for chars
+        import std.utf: byDchar;
 
-    foreach (dchar c; str)
+        return str.byDchar.column(tabsize);
+    }
+    else
     {
-        switch (c)
+        // decoding not needed for wchars and dchars
+        import std.uni : lineSep, paraSep, nelSep;
+
+        size_t column;
+
+        foreach (const c; str)
```

Comment:
const for variables is effectively immutable, and also works when the value being assigned is not immutable.